matching-service/src/controller/matching_service_handlers/check_status_handler.ts

```diff
@@ -74,6 +81,13 @@ class CheckQueueStatusHandler implements
     };
   }
 
+  createRoomToken(queueToken: string): string {
+    const payload: TokenRoomLoad = {
+      room_id: queueToken,
+    };
+    return sign(payload, this.roomSecret);
+  }
+
```

Comment:
This can be better abstracted by extracting it and placing it in something like a `RoomAgent`, and passing the `RoomAgent` to the parts of the code that need to deal with room tokens.